src/main.rs

```diff
@@ -1,238 +1,400 @@
+
+        Swapchain::new(
+            device.clone(),
+            surface.clone(),
+            SwapchainCreateInfo {
+                min_image_count: surface_capabilities.min_image_count,
+
+                image_format,
+
+                image_extent: window.inner_size().into(),
+
+                image_usage: ImageUsage::COLOR_ATTACHMENT,
+
+                composite_alpha: surface_capabilities
+                    .supported_composite_alpha
+                    .into_iter()
+                    .next()
+                    .unwrap(),
 
-    use vulkano::image::ImageUsage;
-    use vulkano::swapchain::{Swapchain, SwapchainCreateInfo};
-
-    // To ensure that only complete images are shown, Vulkan uses what is called a swapchain.
-    // Basically we draw everything that is going to be rendered on a separate screen before displaying it.
-    let (mut swapchain, images) = Swapchain::new(
-        device.clone(),
-        surface.clone(),
-        SwapchainCreateInfo {
-            min_image_count: caps.min_image_count + 1, // How many buffers to use in the swapchain
-            image_format,
-            image_extent: dimensions.into(),
-            image_usage: ImageUsage {
-                color_attachment: true, // What the images are going to be used for
                 ..Default::default()
             },
-            composite_alpha,
-            ..Default::default()
-        },
-    )
-    .unwrap();
-
-    let vertex1 = Vertex {
-        position: [-0.5, -0.5],
-    };
-    let vertex2 = Vertex {
-        position: [0.0, 0.5],
-    };
-    let vertex3 = Vertex {
-        position: [0.5, -0.25],
+        )
+        .unwrap()
     };
 
-    let vertex_buffer = CpuAccessibleBuffer::from_iter(
-        device.clone(),
-        BufferUsage {
-            vertex_buffer: true,
+    let memory_allocator = StandardMemoryAllocator::new_default(device.clone());
+
+    #[derive(BufferContents, Vertex)]
+    #[repr(C)]
+    struct Vertex {
+        #[format(R32G32_SFLOAT)]
+        position: [f32; 2],
+    }
+
+    let vertices = [
+        Vertex {
+            position: [-0.75, -0.20],
+        },
+        Vertex {
+            position: [0.1, 0.83],
+        },
+        Vertex {
+            position: [1.0, -0.1],
+        },
+    ];
+    let vertex_buffer = Buffer::from_iter(
+        &memory_allocator,
+        BufferCreateInfo {
+            usage: BufferUsage::VERTEX_BUFFER,
             ..Default::default()
         },
-        false,
-        vec![vertex1, vertex2, vertex3].into_iter(),
+        AllocationCreateInfo {
+            usage: MemoryUsage::Upload,
+            ..Default::default()
+        },
+        vertices,
     )
     .unwrap();
 
-    let vs = vs::load(device.clone()).expect("failed to create shader module");
-    let fs = fs::load(device.clone()).expect("failed to create shader module");
+    let vs = vs::load(device.clone()).unwrap();
```

Comment:
Isn't the expects nicer? 
https://stackoverflow.com/questions/61301581/when-should-we-use-unwrap-vs-expect-in-rust